### PR TITLE
CPM-786: Delete an identifier from front

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx [hunk 1]
```diff
@@ -16,23 +18,49 @@ type CreateOrEditGeneratorProps = {
...
+  };
+
   return (
     <>
       <Header>
         <PageHeader.Actions>
+          {!isNew && (
+            <SecondaryActions>
+              <SecondaryActions.Item
+                onClick={() => {
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx:55]: a-part function

File: components/identifier-generator/front/src/feature/pages/DeleteGeneratorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      );
+      refetch();
+      onDelete();
+    } catch (error) {
+      notify(NotificationLevel.ERROR, translate('pim_identifier_generator.flash.delete.error'));
+    }
+  };
+
+  return (
+    <Modal closeTitle="Close" illustration={<DeleteIllustration />} onClose={onClose}>
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/DeleteGeneratorModal.tsx:45]: translation

File: components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    jest.spyOn(global, 'fetch').mockResolvedValue({
+      ok: false,
+      statusText: 'An unknown error',
+      json: () => Promise.resolve(),
+    } as Response);
+
+    const onDelete = jest.fn();
+    render(<DeleteGeneratorModal onDelete={onDelete} onClose={jest.fn()} generatorCode={'my_generator'} />);
+
+    expect(screen.getByText('pim_common.delete')).toBeDisabled();
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx:36]: remove

File: components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    fireEvent.change(screen.getByTitle(''), {target: {value: 'my_generator'}});
+    expect(screen.getByText('pim_common.delete')).toBeEnabled();
+    fireEvent.click(screen.getByText('pim_common.delete'));
+
+    await waitFor(() => {
+      return expect(onDelete).toBeCalled();
+    });
+  });
+
+  it('should should not delete when error occured', async () => {
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx:26]: should should

File: components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import {DeleteGeneratorModal} from '../DeleteGeneratorModal';
+import {render, screen, fireEvent, act, waitFor} from '../../tests/test-utils';
+jest.mock('../../hooks/useGetGenerators');
+
+describe('DeleteGeneratorModal', () => {
+  it('should should delete a generator', async () => {
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/__tests__/DeleteGeneratorModal.test.tsx:7]: should should

**Review by pierallard — APPROVED** [review]

**Review by cemorin — APPROVED** [review]

**Review by Beninho — APPROVED** [review]

**Comment by pierallard** [comment]
warmup ?

**Pull request merged**